shared/modules/selectors/smart-transactions.ts

```diff
@@ -56,11 +58,57 @@ type SmartTransactionsMetaMaskState = {
   };
 };
 
-export const getSmartTransactionsOptInStatus = (
-  state: SmartTransactionsMetaMaskState,
-): boolean | null => {
-  return state.metamask.preferences?.smartTransactionsOptInStatus ?? null;
-};
+/**
+ * Returns the user's explicit opt-in status for the smart transactions feature.
+ *
+ * Do not export this selector. It should only be used internally within this file.
+ *
+ * @param state - The state object.
+ * @returns true if the user has explicitly opted in, false if they have opted out,
+ * or null if they have not explicitly opted in or out.
+ */
+const getSmartTransactionsOptInStatusInternal = createSelector(
+  getPreferences,
+  (preferences: {
+    smartTransactionsOptInStatus?: boolean | null;
+  }): boolean | null => {
+    return preferences?.smartTransactionsOptInStatus ?? null;
+  },
+);
+
+/**
+ * Returns the user's explicit opt-in status for the smart transactions feature.
+ * This should only be used for metrics collection, and not for determining if the
+ * smart transactions user preference is enabled.
+ *
+ * To determine if the smart transactions user preference is enabled, use
+ * getSmartTransactionsPreferenceEnabled instead.
+ *
+ * @param state - The state object.
+ * @returns true if the user has explicitly opted in, false if they have opted out,
+ * or null if they have not explicitly opted in or out.
+ */
+export const getSmartTransactionsOptInStatusForMetrics = createSelector(
+  getSmartTransactionsOptInStatusInternal,
+  (optInStatus: boolean | null): boolean | null => optInStatus,
+);
+
+/**
+ * Returns the user's preference for the smart transactions feature.
+ * Defaults to `true` if the user has not set a preference.
+ *
+ * @param state
+ * @returns
+ */
+export const getSmartTransactionsPreferenceEnabled = createSelector(
+  getSmartTransactionsOptInStatusInternal,
+  (optInStatus: boolean | null): boolean => {
+    // In the absence of an explicit opt-in or opt-out,
+    // the Smart Transactions toggle is enabled.
+    const DEFAULT_SMART_TRANSACTIONS_ENABLED = true;
+    return optInStatus ?? DEFAULT_SMART_TRANSACTIONS_ENABLED;
```

Comment:
Rather than defaulting at the selector level, would it be cleaner to change the default state in the `PreferencesController` so the state reflects the preference for new users?

Reply:
Matt, quick question on your comment: 

We currently have three states: default (null), opt-in (true), and opt-out (false). Is there a way to set a default in the preference controller without merging the 'default' and 'opt-in' states? 

The opt-in state isn’t exactly a preference, but we can infer one from it. It probably should have been a separate state property outside of preferences.

Reply:
Unfortunately no, if the state defaults to `true`, you can only determine who has it on or off.

You could update the state to use an enum such as `SmartTransactionOptInStatus` with values like `EnabledByDefault`, `Disabled` and `Enabled` but that's a lot of work.

Though I'm curious what is the value of determining who has explicitly opt-ed in once it's enabled by default?

Suppose the metrics show 100 people turned it on before, 1000 users have it enabled by default, and 200 opted out.

Isn't it just as valuable to know that 1100 people are currently using smart transactions, and 200 opted out?

If a new user toggles the setting, will they become an opt-in rather than a default?